### PR TITLE
Mam improvements

### PR DESCRIPTION
**Pull request by ron282**

### CODE REVIEW

**Comment by ron282** [comment]
Hi. I put Mam improvements is a distinct pull request (without muc improvements).

**Comment by geobra** [comment]
Thank you for splitting the code up in a separate PR. This will make the review much easier. Give me some time for that...

**Comment by ron282** [comment]
Hi regarding omemo, I just saw that there are new versions for lurch, axc, libomemo, mxml... I will check how to integrate them.

**Comment by geobra** [comment]
Okay, need to dig more into MessageHandler.cpp. The other files look good.
From a functional point of view, I get no mam messages into an empty sqlite file. For me, it fails on

if(timestamp > 0 && timestamp <= start)

Which does not have the appropriate values. If I force the if to be true, the mam messages from the group does not go into the group JID, but into the senders JID database.

File: source/base/MessageHandler.cpp [hunk 1]
```diff
@@ -61,52 +61,124 @@ void MessageHandler::handleMessageReceived(Swift::Message::ref message)
...
+        isGroupMessage = true;
+        auto mucUser = message->getPayload<Swift::MUCUserPayload>();
+        if(mucUser)
+        {
+            auto items = mucUser->getItems();
+            if(items.size() > 0 && items[0].realJID)
+            {
+                auto msgOwnerJid = items[0].realJID;
+                direction = msgOwnerJid->compare(clientBareJid, Swift::JID::WithoutResource) == 0 ? 0 : 1;
+                partyJid = QString::fromStdString(msgOwnerJid->toBare().toString());
```
Comment (geobra) [inline source/base/MessageHandler.cpp:150]:
```suggestion
                partyJid = QString::fromStdString(message->getFrom().toBare().toString());
```

Comment (geobra) [reply source/base/MessageHandler.cpp:150]: This fixes the issue for me, that the mam group messages are not in the group room, but associated to the sender of the message.

**Comment by geobra** [comment]
@ron282  Do you still want to work on that? If so, take your time, no hurry ;-). I just want to check if I should bring this PR to a final state, or if you want to do this.

**Comment by ron282** [comment]
@geobra Yes I want but I don't have a lot of time. May be in May!

**Comment by geobra** [comment]
Perfect - works for me ;-).

**Pull request closed without merging**